keps/2724-topology-aware-scheduling/README.md

```diff
@@ -743,6 +747,65 @@ to ungate a Pod. The expectation is fulfilled if the Pod is observed as ungated
 or the ungating request fails. We hold ungating if there are pending ungatings
 within the PodSet.
 
+### Support for ProvisioningRequests
+
+We are going to support autoscaling via ProvisioningRequest AdmissionCheck.
+
+The key assumptions of the solution:
+- When there is ProvisioningRequest on the list of AdmissionChecks Kueue
+  Scheduler only reserves quota, but skips TAS assignment until all
+  AdmissionChecks (including the ProvisioningRequest) are `Ready` (green).
+- When all AdmissionChecks are `Ready` Kueue Scheduler does "second pass" on the
+  workload to extend the "TAS" assignment, but respecting the previously
+  selected resource flavor.
+- In order to do "second pass" for such workloads Kueue Scheduler maintains a
+  dedicated "secondPassQueue" queue of workloads ready for the second pass -
+  with QuotaReservation and all AdmissionChecks `Ready`, but without required
+  TopologyAssignments.
+
+#### Determining the need for second pass
+
+In order to determine if the "second pass" is needed, and thus if
+`TopologyAssignment` is required, Kueue sets the following field as true
+at the moment of QuotaReservation (first pass).
+
+```golang
+type PodSetAssignment struct {
+  ...
+  // topologyAssignmentRequested indicates the topology assignment was requested
+  // at the moment of quota reservation. Workloads for which the value of the
+  // field is true, but `TopologyAssignment` is nil as subject for the second
+  // pass of scheduling before admission to determine the TopologyAssignment.
+  //
+  // +optional
+  TopologyAssignmentRequested *bool `json:"topologyAssignmentRequested,omitempty"`
```

Comment:
Actually, I'm still wondering. Probably just a condition "WaitingForSecondPass" would be enough.

Reply:
Actually, I updated the KEP to introduce `delayedTopologyRequest`. I tried to introduce `SecondPassRequired` but found it difficult. The new field is only set in the new scenarios which makes the API fully backwards compatible.